File: robocode-tankroyale-bot-api-csharp/src/TurnCompleteCondition.cs
using Robocode.TankRoyale.BotApi;
using Robocode.TankRoyale.BotApi.Events;

public class TurnCompleteCondition : Condition
{
  private readonly Bot bot;

  public TurnCompleteCondition(Bot bot)
  {
    this.bot = bot;
  }

  public override bool Test()
  {
    return bot.TurnRemaining == 0;
  }
}

File: robocode-tankroyale-bot-api-java/src/main/java/dev/robocode/tankroyale/bot/TurnCompleteCondition.java
package dev.robocode.tankroyale.bot;

import dev.robocode.tankroyale.botapi.Bot;
import dev.robocode.tankroyale.botapi.Condition;
import dev.robocode.tankroyale.botapi.IBot;

public class TurnCompleteCondition extends Condition {

    private final Bot bot;

    public TurnCompleteCondition(Bot bot) {
        this.bot = bot;
    }

    @Override
    public boolean test() {
        return bot.getTurnRemaining() == 0;
    }
}
